pkg/apis/core/v1beta1/helper/errors.go

```diff
@@ -65,47 +68,59 @@ func DetermineError(err error, message string) error {
 		errMsg = err.Error()
 	}
 
-	code := determineErrorCode(err)
-	if code == "" {
+	codes := determineErrorCodes(err)
+	if codes == nil {
 		return errors.New(errMsg)
 	}
-	return &errorWithCode{code, errMsg}
+	return &errorWithCodes{errMsg, codes}
 }
 
-func determineErrorCode(err error) gardencorev1beta1.ErrorCode {
-	var coder Coder
+func determineErrorCodes(err error) []gardencorev1beta1.ErrorCode {
+	var (
+		coder   Coder
+		message = err.Error()
+		codes   = sets.NewString()
+	)
 
-	// first try to re-use code from error
+	// try to re-use code from error
 	if errors.As(err, &coder) {
-		switch coder.Code() {
-		case gardencorev1beta1.ErrorInfraUnauthorized, gardencorev1beta1.ErrorInfraQuotaExceeded, gardencorev1beta1.ErrorInfraInsufficientPrivileges, gardencorev1beta1.ErrorInfraDependencies, gardencorev1beta1.ErrorInfraResourcesDepleted, gardencorev1beta1.ErrorConfigurationProblem:
-			return coder.Code()
+		for _, code := range coder.Codes() {
+			codes.Insert(string(code))
 		}
 	}
 
-	message := err.Error()
-	switch {
-	case unauthorizedRegexp.MatchString(message):
-		return gardencorev1beta1.ErrorInfraUnauthorized
-	case quotaExceededRegexp.MatchString(message):
-		return gardencorev1beta1.ErrorInfraQuotaExceeded
-	case insufficientPrivilegesRegexp.MatchString(message):
-		return gardencorev1beta1.ErrorInfraInsufficientPrivileges
-	case dependenciesRegexp.MatchString(message):
-		return gardencorev1beta1.ErrorInfraDependencies
-	case resourcesDepletedRegexp.MatchString(message):
-		return gardencorev1beta1.ErrorInfraResourcesDepleted
-	case configurationProblemRegexp.MatchString(message):
-		return gardencorev1beta1.ErrorConfigurationProblem
-	default:
-		return ""
+	// determine error codes
+	if unauthorizedRegexp.MatchString(message) {
+		codes.Insert(string(gardencorev1beta1.ErrorInfraUnauthorized))
 	}
+	if quotaExceededRegexp.MatchString(message) {
+		codes.Insert(string(gardencorev1beta1.ErrorInfraQuotaExceeded))
+	}
+	if insufficientPrivilegesRegexp.MatchString(message) {
+		codes.Insert(string(gardencorev1beta1.ErrorInfraInsufficientPrivileges))
+	}
+	if dependenciesRegexp.MatchString(message) {
+		codes.Insert(string(gardencorev1beta1.ErrorInfraDependencies))
+	}
+	if resourcesDepletedRegexp.MatchString(message) {
+		codes.Insert(string(gardencorev1beta1.ErrorInfraResourcesDepleted))
+	}
+	if configurationProblemRegexp.MatchString(message) {
+		codes.Insert(string(gardencorev1beta1.ErrorConfigurationProblem))
+	}
+
+	// eliminate duplicates
```

Comment:
What do you mean by that? Duplicates are already eliminated by using a `set`, right?